frontend/src/metabase/lib/time.js

```diff
@@ -40,9 +43,39 @@ const NUMERIC_UNIT_FORMATS = {
       .startOf("quarter"),
 };
 
+/// Update Moment locale information such as the first day of the week based on their values in the Metabase instance
+/// settings.
+function updateLocaleInfo() {
+  const startOfWeekDayName = MetabaseSettings.get("start-of-week");
+  if (!startOfWeekDayName) {
+    return;
+  }
+
+  const START_OF_WEEK_DAYS = [
+    "sunday",
+    "monday",
+    "tuesday",
+    "wednesday",
+    "thursday",
+    "friday",
+    "saturday",
+  ];
+  const startOfWeekDayNumber =
+    _.indexOf(START_OF_WEEK_DAYS, startOfWeekDayName) || 0;
```

Comment:
`indexOf` exists on `Array`: `START_OF_WEEK_DAYS.indexOf(startOfWeekDayName)`.

Both underscore's and the built in methods return `-1` when not found, so `|| 0` won't work unfortunately.